undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java

```diff
@@ -949,19 +954,6 @@ public void resume() {
                     });
                     suspendController.get().registerActivity(serverActivity.get());
                 });
-                ServletContextListener sl = new ServletContextListener() {
-                    @Override
-                    public void contextInitialized(ServletContextEvent sce) {}
-
-                    @Override
-                    public void contextDestroyed(ServletContextEvent sce) {
-                        final ServerActivity activity = serverActivity.get();
-                        if(activity != null) {
-                            suspendController.get().unRegisterActivity(activity);
-                        }
-                    }
-                };
-                d.addListener(new ListenerInfo(sl.getClass(), new ImmediateInstanceFactory<EventListener>(sl)));
```

Comment:
I don't know enough about this code to know if removing this is ok, but if there are concerns from those who do I think you can achieve your objective by leaving this here and replacing serverActivity.get() both here and in the stop method with serverActivity.getAndSet(null).

Reply:
@aogburn / @fl4via  can you check this one, it looks like the fix has been stalled for several months

Reply:
yes, it is okay to remove this block.